tests/test_system_health.py

```diff
@@ -0,0 +1,87 @@
+"""Test system health."""
+
+import asyncio
+from typing import Any, Generator
+
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+import pytest
+
+from custom_components.hacs.base import HacsBase
+
+from tests.common import MockedResponse, ResponseMocker, safe_json_dumps
+from tests.conftest import SnapshotFixture
+
+
+HACS_SYSTEM_HEALTH_DOMAIN = "Home Assistant Community Store"
+
+
+async def get_system_health_info(hass: HomeAssistant, domain: str) -> dict[str, Any]:
+    """Get system health info."""
+    return await hass.data["system_health"][domain].info_callback(hass)
+
+
+async def test_system_health(
+    setup_integration: Generator,
+    hass: HomeAssistant,
+    response_mocker: ResponseMocker,
+    snapshots: SnapshotFixture,
+) -> None:
+    """Test HACS system health."""
+    response_mocker.add(
+        url="https://api.github.com",
+        response=MockedResponse(
+            content={},
+            headers={"Content-Type": "application/json"},
+        ),
+    )
+    response_mocker.add(
+        url="https://raw.githubusercontent.com/hacs/integration/main/hacs.json",
+        response=MockedResponse(
+            content={},
+            headers={"Content-Type": "application/json"},
+        ),
+    )
+    response_mocker.add(
+        url="https://data-v2.hacs.xyz/data.json",
+        response=MockedResponse(
+            content={},
+            headers={"Content-Type": "application/json"},
+        ),
+    )
```

Comment:
I don't think we need real fixtures for these, system health just checks if the URLs are reachable